packages/main/src/chat/chat-manager.ts

```diff
@@ -64,39 +67,39 @@ export class ChatManager {
     return userMessages.at(-1);
   }
 
-  async streamText(
-    _listener: Electron.IpcMainInvokeEvent,
-    providerId: string,
-    connectionName: string,
-    modelId: string,
-    mcp: Array<string>,
-    messages: UIMessage[],
-    onDataId: number,
-  ): Promise<number> {
-    const internalProviderId = this.providerRegistry.getMatchingProviderInternalId(providerId);
-    const sdk = this.providerRegistry.getInferenceSDK(internalProviderId, connectionName);
-    const languageModel = sdk.languageModel(modelId);
+  private async getInferenceComponents(
+    params: InferenceParameters,
+  ): Promise<Parameters<typeof streamText>[0] & Parameters<typeof generateText>[0]> {
```

Comment:
return type is pretty unclear to me, can we add some comments on the function, to explain _shortly_ what it is doing?